### PR TITLE
[FABCN-408] Separate message handler from client

### PR DESCRIPTION
**Pull request by shimos**
This patch divides the ChaincodeSupportClient class into two classes:
ChaincodeSupportClient, which represents a chaincode client that
connects to a peer, and ChaincodeMessageHandler, which handles messages
in communciation with the peer through a stream.

Because the latter part is common in both client and server model,
the ChaincodeMessageHandler class will be used in the future
implementation for a chaincode gRPC server.

Signed-off-by: Taku Shimosawa <taku.shimosawa@hal.hitachi.com>

### CODE REVIEW

File: libraries/fabric-shim/lib/handler.js [hunk 1]
```diff
@@ -527,17 +553,6 @@ class ChaincodeSupportClient {
             this.msgQueueHandler.queueMsg(qMsg);
         });
     }
-
```
Comment (mbwhite) [inline libraries/fabric-shim/lib/handler.js:530]: would prefer not to remove the toString as this will help with log files and debugging.

**Comment by shimos** [comment]
@mbwhite Thank you for your review. I'm going to fix the part you have pointed out.

As in the commit title, https://jira.hyperledger.org/browse/FABCN-408 is the JIRA issue.
This work is intended for implementation of the chaincode as a server in node.js chaincode.

**Comment by mbwhite** [comment]
> @mbwhite Thank you for your review. I'm going to fix the part you have pointed out.
> 
> As in the commit title, https://jira.hyperledger.org/browse/FABCN-408 is the JIRA issue.
> This work is intended for implementation of the chaincode as a server in node.js chaincode.

Thanks... do you think it will be complete in the next few weeks? If so it could be part of the next release of the release-2.x branch... If so could you cherry-pick to that branch please.

**Comment by shimos** [comment]
@mbwhite Thank you for merging. I think I can complete this within this month. Once most work is finished, I will submit PR for release-2.x.

**Pull request merged**